### PR TITLE
Adding vector serialization and deserialization functions for PyVelox

### PR DESCRIPTION
**Pull request by vibhatha**
This PR adds support for `VectorSaver.h` functionality for serializing and deserializing vectors.

`VectorSaver.h` includes `saveVectorToFile` and `restoreVectorFromFile` methods. This PR includes bindings for those methods as `save_vector` and `load_vector`.

### CODE REVIEW

File: pyvelox/pyvelox.cpp [hunk 1]
```diff
@@ -119,6 +119,26 @@ static void addExpressionBindings(
       });
 }
 
+static VectorPtr pyRestoreVectorFromFileHelper(
```
[the remark this answers is not in the document]
Comment (vibhatha) [reply pyvelox/pyvelox.cpp:122]: Sure. I will update.

File: pyvelox/test/test_vector.py [hunk 2]
```diff
@@ -254,3 +256,22 @@ def test_append(self):
...
+    def setUp(self):
+        # create a temporary directory
+        self.test_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        # remove the temporary directory
+        shutil.rmtree(self.test_dir)
+    
+    def test_serde_vector(self):
+        vec = pv.from_list([1, 2, 3])
```
Comment (kgpai) [inline pyvelox/test/test_vector.py:271]: Can you add a test for different encodings of vectors : 
1 - Constant 
2 - Dictionary

Also can you add string and float to this ?

**Review by kgpai — COMMENTED** [review]
Looks great, just a little more additional tests.

File: pyvelox/serde.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+} // namespace
+
+void addSerdeBindings(py::module& m, bool asModuleLocalDefinitions) {
+  using namespace facebook::velox;
+
+  m.def(
+      "save_vector",
+      &saveVectorToFile,
+      R"delimiter(
+        Serializes the vector into binary format and writes it to a new file.
```
Comment (kgpai) [inline pyvelox/serde.cpp:42]: Thanks , this is so nice to have !

File: pyvelox/test/test_serde.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        for vec_key, fpath_key in zip(data, paths):
+            vec = data[vec_key]
+            fpath = paths[fpath_key]
+            pv.save_vector(vector=vec, file_path=fpath)
+            loaded_vec = pv.load_vector(file_path=fpath)
+            self.assertEqual(len(vec), len(loaded_vec))
+            for i in range(len(vec)):
+                self.assertEqual(vec[i], loaded_vec[i])
```
Comment (kgpai) [inline pyvelox/test/test_serde.py:61]: This only checks the values, can we also ensure encoding doesnt change, i.e constant vector remains constant ?

**Review by kgpai — COMMENTED** [review]
Looks great, one minor change though.

**Comment by vibhatha** [comment]
@kgpai added a test for type checking. Let's wait for the CIs.

**Review by kgpai — APPROVED** [review]

**Comment by vibhatha** [comment]
@kgpai I see an internal linter is failing. How should we fix it?

**Pull request closed without merging**